src/logic/hooks/useEstimateSafeCreationGas.tsx

```diff
@@ -27,11 +29,14 @@ const estimateGas = async (
   safeCreationSalt: number,
   addresses: string[],
 ): Promise<SafeCreationEstimationResult> => {
-  const [gasEstimation, gasPrice] = await Promise.all([
+  const [gasEstimation, gasPrice, feesPerGas] = await Promise.all([
     estimateGasForDeployingSafe(addresses, numOwners, userAccount, safeCreationSalt),
     calculateGasPrice(),
+    getFeesPerGas(),
   ])
+
   const estimatedGasCosts = gasEstimation * parseInt(gasPrice, 10)
+  const maxPrioFeePerGas = setMaxPrioFeePerGas(feesPerGas.maxPriorityFeePerGas, estimatedGasCosts)
```

Comment:
`estimatedGasCosts` needs to be replaced with `gasPrice` as its the per unit cost we want to compare with

Reply:
Why? The `estimateGasCosts` is calculated from `gasPrice`. See line 38.